### PR TITLE
Renamed isnan() to IsReasonable

### LINKED ISSUE #603: Unusual definition of isnan() and its implications
Commit ea6db67c7 added an unusual `isnan` macro:

```c++
#define isnan(x) (((x) != (x)) || (x > 1e11) || (x < -1e11))
```

Commit 8bc322eb4 adds a preprocessor guard that *looks* like it would cause the `isnan` function from `math.h` to be preferred, but doesn't actually do that on many platforms, e.g. glibc:

```c++
#ifndef isnan
#   define isnan(x) (((x) != (x)) || (x > 1e11) || (x < -1e11))
#endif
```

@jwesthues I'm not really sure what's the intent here--is it to check for NaN and infinity, or is it really a comparison with `VERY_POSITIVE` and `VERY_NEGATIVE`? Regardless, we need a better name for the function that doesn't clash with `math.h`, and to get rid of the misleading preprocessor junk.

### PR DESCRIPTION
**Pull request by phkahler**
Fixes #603.

### CODE REVIEW

File: src/solvespace.h [hunk 1]
```diff
@@ -76,8 +76,8 @@ typedef struct _cairo_surface cairo_surface_t;
     dbp("tri: (%.3f %.3f %.3f) (%.3f %.3f %.3f) (%.3f %.3f %.3f)", \
         CO((tri).a), CO((tri).b), CO((tri).c))
 
-#ifndef isnan
-#   define isnan(x) (((x) != (x)) || (x > 1e11) || (x < -1e11))
+#ifndef IsReasonable
+#   define IsReasonable(x) (((x) != (x)) || (x > 1e11) || (x < -1e11))
```
Comment (whitequark) [inline src/solvespace.h:80]: The whole point of renaming would be to get rid of a name collision and all this preprocessor junk. This function should be `static inline` inside `namespace SolveSpace`, like the `WRAP` ones.

Comment (phkahler) [reply src/solvespace.h:80]: Oh, ok.

Comment (phkahler) [reply src/solvespace.h:80]: You want the WRAP one to be static too?

Comment (whitequark) [reply src/solvespace.h:80]: Ah, I misremembered that it was already static. It makes zero difference either way.

Comment (phkahler) [reply src/solvespace.h]: Done.  unless you want the 3 wrap functions inline too.

**Comment by whitequark** [comment]
@phkahler I applied the rename in c1cd315d. I did it myself because in this PR you did not include either the issue number or the explanation of the change, so for anyone reading the history it would be a mystery why it was done.

**Pull request closed without merging**